lib/orderbook/errors.ts

```diff
@@ -45,6 +46,10 @@ const errors = {
     message: `order with id ${orderId} could not be found`,
     code: errorCodes.ORDER_NOT_FOUND,
   }),
+  UNAUTHORIZED: (orderId: string, peerPubKey: string) => ({
+    message: `order with id ${orderId} does not belong to ${peerPubKey}`,
```

Comment:
I'd be inclined to either rename this error to something like `UNAUTHORIZED_INVALIDATION` or to make the message more generic, like `peer attempted to perform an unauthorized action`.